File: platform/newservice/k8s_api_client/api_client/response_code/code.py
# -*- coding: utf-8 -*-
# Author: YanHua <it-yanh@all-reach.com>

# 程序执行状态码定义,分类说明如下：
# 0      OK 只有返回码为0才有result值。
# 1xx    API接口自身错误，如参数错误、接口已停用、找不到接口。
# 2xx    认证或权限错误
# 3xx    逻辑错误，如名称冲突，资源超过限额，余额不足等。
# 4xx    数据库错误
# 5xx    驱动层错误
# 6xx    系统级别错误

status_code = {
         0:   "OK",
        101: "Parameters error",
        201: "Authentication failure",
        202: "Operation denied",
        301: "Resource name already exists",
        401: "Database insert error",
        402: "Database delete error",
        403: "Database update error",
        404: "Database select error",
        405: "did not have this resource",
        501: "kubernetes resource create failure",
        502: "kubernetes resource update failure",
        503: "kubernetes resource delete failure",
        504: "kubernetes resource select failure",
        505: "kubernetes resource is creating...",
        511: "Ceph disk create failure",
        512: "Ceph disk delete failure",
        513: "Ceph disk resize failure"

}


def request_result(code, ret={}):

    result = {
                 "status": code,
                 "msg": status_code[code],
                 "result": ret
             }

    return result

File: platform/other/livenowhy/dataModel/code.go
package dataModel

import (
	"encoding/json"
	"fmt"
	"io"
	"net/http"
)

var StatusCode = make(map[int]string)
func init() {
	StatusCode[0] = "OK"
	StatusCode[101] = "Parameters error"
	StatusCode[102] = "RPC API routing error"
	StatusCode[103] = "Http requests error"
	StatusCode[201] = "Authentication failure"
	StatusCode[202] = "Operation denied"
	StatusCode[301] = "Resource name already exists"
	StatusCode[401] = "Database insert error"
	StatusCode[402] = "Database delete error"
	StatusCode[403] = "Database update error"
	StatusCode[404] = "Database select error"
	StatusCode[501] = "Kubernetes resource create failure"
	StatusCode[502] = "Kubernetes resource update failure"
	StatusCode[503] = "Kubernetes resource delete failure"
	StatusCode[511] = "Ceph disk create failure"
	StatusCode[512] = "Ceph disk delete failure"
	StatusCode[513] = "Ceph disk resize failure"
	StatusCode[597] = "RabbitMQ rpc client exec timeout"
	StatusCode[598] = "RabbitMQ rpc client exec error"
	StatusCode[599] = "RabbitMQ rpc server exec error"
	StatusCode[601] = "System error"
	StatusCode[701] = "There is no resources"
	StatusCode[702] = "ERROR"
	StatusCode[703] = "no method"
}

type ResponseResult struct {
	Status int         `json:"status"`
	Result interface{} `json:"result"`
	Msg    string      `json:"msg"`
}

func RequestResult(code int, ret interface{}) (rel ResponseResult) {
	result := ResponseResult{
		Status: code,
		Msg:    StatusCode[code],
		Result: ret,
	}
	return result
}





func HttpRequestResult(w http.ResponseWriter, code int, ret interface{}) {
	request := ResponseResult{
		Status: code,
		Msg: StatusCode[code],
		Result: ret,
	}
	w.Header().Set("Content-Type", "application/json")

	s, err := json.Marshal(request)
	if err != nil {
		fmt.Errorf("HttpRequestResult is error : %s", err.Error())
	}
	io.WriteString(w, string(s))
}


func HttpRequestResultString(code int, ret interface{}) (string) {
	request := ResponseResult{
		Status: code,
		Msg: StatusCode[code],
		Result: ret,
	}
	s, err := json.Marshal(request)
	if err != nil {
		fmt.Errorf("HttpRequestResult is error : %s", err.Error())
	}
	return string(s)
}